app/Http/Middleware/LocationDetector.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\Services\Geolocator\ClientIpInfo;
+use Closure;
+use Illuminate\Http\Request;
+
+class LocationDetector
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $user = $request->user();
+        $userClintIp = \Request::ip();
+        if ($userClintIp == '127.0.0.1')
```

Comment:
Please put the braces this does not comply with PSR, also make sure to run `composer pint` before you commit